src/extension/background-script/actions/webbtc/getAddress.ts

```diff
@@ -0,0 +1,60 @@
+import { getTaprootAddressFromPrivateKey } from "~/common/lib/btc";
+import { decryptData } from "~/common/lib/crypto";
+import {
+  BTC_TAPROOT_DERIVATION_PATH,
+  BTC_TAPROOT_DERIVATION_PATH_REGTEST,
+  derivePrivateKey,
+  derivePublicKey,
+} from "~/common/lib/mnemonic";
+import state from "~/extension/background-script/state";
+import { BitcoinAddress, MessageGetAddress } from "~/types";
+
+const getAddress = async (message: MessageGetAddress) => {
+  try {
+    // TODO: is this the correct way to decrypt the mnmenonic?
+    const password = await state.getState().password();
+    if (!password) {
+      throw new Error("No password set");
+    }
+    const account = await state.getState().getAccount();
+    if (!account) {
+      throw new Error("No account selected");
+    }
+    if (!account.mnemonic) {
+      throw new Error("No mnemonic set");
+    }
+    const mnemonic = decryptData(account.mnemonic, password);
+    const settings = state.getState().settings;
+
+    const derivationPath =
+      settings.bitcoinNetwork === "bitcoin"
+        ? BTC_TAPROOT_DERIVATION_PATH
+        : BTC_TAPROOT_DERIVATION_PATH_REGTEST;
+
+    const privateKey = derivePrivateKey(mnemonic, derivationPath);
+    const publicKey = derivePublicKey(mnemonic, derivationPath);
+
+    const address = getTaprootAddressFromPrivateKey(
```

Comment:
I have a bit of an object oriented programming mind...
so maybe we can have some key object that we load from the state which we then can ask for derived keys?
then we don't need to handle decryption of the mnemonic in the various places and also don't need to pass in the mnemonic to those functions. 

e.g. 

```
const key = state.getState().getAccountKey(); 
const derivedKey = key.deriveKey(derivationPath);
derivedKey.publicKey;
```

those methods here are also rather similar: https://github.com/getAlby/lightning-browser-extension/pull/2338/files#diff-7d7f5ebb783c099012b8a0f530ba786ceae995894f55dba8b57f0a6834592a9dR13-R30

and once we have the key derived we can get the public key from that without the need to fully derive it again?

Reply:
@bumi sounds good to me. I will create a separate PR for this change. Maybe we should call this method `getSecretKey()` to maintain the same terminology?